test/models/deploys_test.rb

```diff
@@ -325,6 +325,56 @@ def setup
     end
   end
 
+  test "#confirm! bails out if the deploy is successful already" do
+    assert_predicate @deploy, :success?
+
+    Deploy::CONFIRMATIONS_REQUIRED.times do
+      @deploy.confirm!
+      assert_predicate @deploy, :success?
+    end
+  end
+
+  test "#confirm! first transition to flapping then ultimately to success if the deploy was failed" do
+    @deploy = deploys(:shipit2)
+    assert_predicate @deploy, :failed?
+
+    @deploy.confirm!
+    assert_predicate @deploy, :flapping?
+
+    (Deploy::CONFIRMATIONS_REQUIRED - 2).times do
```

Comment:
why not make this `-1` instead and get rid of the first iteration above? it looks identical